pydra/engine/node.py

```diff
@@ -236,6 +240,12 @@ def cache_dir(self, location):
             self._cache_dir = mkdtemp()
             self._cache_dir = Path(self._cache_dir)
 
+    @property
+    def is_runnable(self):
```

Comment:
@mgxd - this is the kind of thing i would add a function to check and use the function in the workflow class. the function can take a set of dependencies (this will come in handy when you start dealing with splits).

try keeping functions out of classes when it's not necessary. also, please do test, because adding task pointers may make the class not pickle properly.